### PR TITLE
[WIP] pytest-apm second iteration

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

* add a new python step to execute Python code in a virtual environment.
* add withAPMService step to encapsulate an APM service access.
* add withPytestAPM step to run pytest test reporting APM traces.

### CODE REVIEW

File: vars/apmCli.groovy [hunk 1]
```diff
@@ -31,7 +28,7 @@ def call(Map args = [:]) {
   def spanLabel = args.containsKey('spanLabel') ? args.spanLabel : ''
   def result = args.containsKey('result') ? args.result : ''
 
-  if(!isUnix() || !serviceName){
+  if(!serviceName){
```
Comment (kuisathaverat) [inline vars/apmCli.groovy:31]: now should be multi-os

File: vars/apmCli.groovy [hunk 2]
```diff
@@ -44,20 +41,9 @@ def call(Map args = [:]) {
...
-    writeFile file: "apm-cli.py", text: apmCliPy
-
-    if(!url && !token){
-      def apmJson = getVaultSecret(secret: "${apmCliConfig}")?.data.value
-      def apm = readJSON(text: apmJson)
-      url = apm.url
-      token = apm.token
-    }
-
+    def apmCliRelease = "https://github.com/elastic/apm-pipeline-library/releases/download/apm-cli-0.0.1/apm-cli-0.0.1.tar.gz"
```
Comment (kuisathaverat) [inline vars/apmCli.groovy:45]: it directly installs the release from GitHub

File: vars/apmCli.groovy [hunk 3]
```diff
@@ -70,36 +56,23 @@ def call(Map args = [:]) {
...
+        "APM_CLI_TRANSACTION_NAME=${transactionName}",
+        "APM_CLI_SPAN_NAME=${spanName}",
+        "APM_CLI_SPAN_COMMAND=${spanCommand}",
+        "APM_CLI_SPAN_LABELS=${spanLabel}",
+        "APM_CLI_PARENT_TRANSACTION_SAVE=${ saveTsID ? 'tsID.txt' : ''}",
+        "APM_CLI_TRANSACTION_RESULT=${result}",
+        "APM_CLI_PARENT_TRANSACTION=${parentTransaction}",
+      ]){
+        log(level: 'DEBUG', text: "apmCli: Runninf APM CLI")
+        python(label: 'apmCLI', file: 'apm-cli.py')
```
Comment (kuisathaverat) [inline vars/apmCli.groovy:71]: less code :D

File: vars/apmCli.groovy (hunk 2 above)
Comment (v1v) [reply vars/apmCli.groovy:45]: I'd say to add the `retryWithSleep` step

File: vars/apmCli.groovy (hunk 3 above)
Comment (v1v) [reply vars/apmCli.groovy:71]: should this particular step trap the errors to avoid failing the build if something bad happened?

**Comment by kuisathaverat** [comment]
we will use the Otel version

**Pull request closed without merging**